scripts/run-if-package-requested.js

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env node
+'use strict';
+const path = require('path');
+const packageDir = path.basename(process.cwd());
+if (process.env.MONGOSH_RUN_ONLY_IN_PACKAGE && process.env.MONGOSH_RUN_ONLY_IN_PACKAGE !== packageDir) {
+  process.stderr.write(
+    `${packageDir} is not ${process.env.MONGOSH_RUN_ONLY_IN_PACKAGE}, skipping "${process.argv.slice(2).join(' ')}"\n`);
+  return;
+}
+const child_process = require('child_process');
+child_process.spawn(process.argv[2], process.argv.slice(3), { stdio: 'inherit', shell: process.platform === 'win32' })
```

Comment:
Do you know why the `shell:` flag is required here?

Reply:
Ah yeah when I tried it without it, it didn't seem to properly pass on the current environment (like `$PATH` etc.) as it would fail to launch `mocha` for example.

Reply:
Ah, I see … I think this is the https://www.npmjs.com/package/cross-spawn problem, if you like, feel free to use that, but as long as we don’t accept arbitrary inputs here I think this is fine as well